blog.qmd

```diff
@@ -0,0 +1,329 @@
+
+To install and load the required packages, we can use the librarian package for
+convenience
+
+```{r text-}
+librarian::shelf(tidyverse, # data wrangling
+                 quanteda, # text mining
+                 readtext # read texts and associated document-level meta-data
+)
+```
+
+Next, we load our text data containing course descriptions as well as
+document-level metadata using the `readtext::readtext()` function. We then
+perform some additional cleaning steps to standardize variable names and focus
+our analysis on active courses for credit. Here's a brief explanation of each
+column in our courses dataframe:
+
+-   doc_id: This is a unique identifier for each course. It is constructed by
+    combining the school abbreviation, the course prefix, and the course number.
+    For example, 'BC-THE-2300' refers to a course at the school abbreviated as
+    'BC', with the course prefix 'THE', and the course number '2300'.
+
+-   discipline: This column contains the discipline or subject area of the
+    course. The discipline is represented as a code and a description, such as
+    '080 - THEATRE ARTS'.
+
+-   course_title: This is the title of the course.
+
+-   course_credits: This column indicates the number of credit hours that a
+    student receives upon completion of the course.
+
+-   text: This column contains the course description text that was scraped from
+    the [Florida Department of Education's course catalog
+    website](https://flscns.fldoe.org/Default.aspx). This is the main body of
+    text that will be analyzed using text mining techniques. The text appears to
+    be truncated in the provided data, as indicated by the ellipsis at the end
+    of each string.
+
+```{r}
+courses <-
+  readtext(here::here("data/data-intermediate/course-descriptions.csv"), 
+           text_field = "course_description",
+           docid_field = 'course_id') %>% 
+  janitor::clean_names() %>% 
+  filter(course_status == "ACTIVE",
+         type_of_credit == "COLLEGE CREDIT") %>% 
+  select(doc_id, discipline, course_title, course_credits, text) %>%
+  mutate(course_title = str_to_title(course_title),
+         discipline = str_to_title(discipline),
+         text = str_to_sentence(text))
+```
+
+```{r}
+#| echo: false
+courses %>% 
+  slice(1:5) %>% 
+  gt() %>% 
+    opt_interactive(
+    use_resizers = TRUE,
+    use_filters = TRUE,
+    use_compact_mode = TRUE,
+    use_highlight = TRUE,
+    use_search = TRUE,
+    use_text_wrapping = TRUE
+  ) %>%
+    cols_width(
+    doc_id ~ px(150),
+    discipline ~ px(180),
+    course_title ~ px(150),
+    course_credits ~ px(80),
+    text ~ px(350)
+  ) %>% 
+  urbn_gt_theme()
+```
+
+The first step in our analysis is to create a corpus, a collection of text
+documents, from our course descriptions. We can achieve this using the
+`corpus()` function in `quanteda.` Next, we extract the tokens in the
+corpus---usually words, but they can also be n-grams or multi-word expressions.
```

Comment:
Still think a link to definition of n-grams would be helpful here!

Reply:
Added a link